contracts/test/integration/pcv/CurveSwapper.t.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity ^0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {Vm} from "../../utils/Vm.sol";
+import {DSTest} from "../../utils/DSTest.sol";
+import {CurveSwapper} from "../../../pcv/curve/CurveSwapper.sol";
+import {Core} from "../../../core/Core.sol";
+import {MainnetAddresses} from "../fixtures/MainnetAddresses.sol";
+
+contract CurveSwapperIntegrationTest is DSTest {
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+    address payable receiver = payable(address(3));
+
+    Core private core = Core(MainnetAddresses.CORE);
+
+    CurveSwapper private curveSwapper;
+
+    // dummy function to avoid errors
+    // the CurveSwapper will call deposit() on the
+    // tokenReceivingAddress after performing a swap
+    function deposit() public {}
```

Comment:
Feels strange to me to rely on this test file also being an 'active contract/mock' in your tests. Why not create a simple dummy mock contract at the top of this file with an empty `deposit()` function, deploy that and then set that address to be the `tokenReceivingAddress?`